File: python-algo/leetcode/problem_557.py
"""557. Reverse Words in a String III
https://leetcode.com/problems/reverse-words-in-a-string-iii/

Given a string, you need to reverse the order of characters in each word
within a sentence while still preserving whitespace and initial word order.

Example 1:

Input: "Let's take LeetCode contest"
Output: "s'teL ekat edoCteeL tsetnoc"

Note: In the string, each word is separated by single space and there will
not be any extra space in the string.
"""


class Solution:
    def reverse_words(self, s: str) -> str:
        words = s.split(' ')
        new_words = []
        for word in words:
            new_words.append(word[::-1])
        return ' '.join(new_words)

File: java-algo/src/main/java/com/leetcode/Problem557.java
package com.leetcode;

/**
 * 557. Reverse Words in a String III
 * https://leetcode.com/problems/reverse-words-in-a-string-iii/
 */
class Problem557 {
    public String reverseWords(String s) {
        StringBuilder ans = new StringBuilder();
        String[] words = s.split(" ");
        for (int i = 0; i < words.length; i++) {
            String rev = reverse(words[i]);
            ans.append(rev);
            if (i != words.length - 1) {
                ans.append(" ");
            }
        }
        return ans.toString();
    }

    private String reverse(String word) {
        StringBuilder sb = new StringBuilder();
        for (int i = word.length() - 1; i >= 0; i--) {
            sb.append(word.charAt(i));
        }
        return sb.toString();
    }
}
